src/constants.ts

```diff
@@ -1,4 +1,39 @@
+import { Enemy } from './enemy';
+import { Player } from './player';
+import { Weapon } from './weapon';
+
 export enum CanvasValues {
-  Width = 800,
-  Height = 600,
+  WIDTH = 800,
+  HEIGHT = 600,
 }
+
+export enum FacingAngles {
+  TOP = Math.PI,
+  BOTTOM = 0,
+  LEFT = (90 * Math.PI) / 180,
+  RIGHT = (270 * Math.PI) / 180,
+  TOP_LEFT = (135 * Math.PI) / 180,
+  TOP_RIGHT = (225 * Math.PI) / 180,
+  BOTTOM_LEFT = (45 * Math.PI) / 180,
+  BOTTOM_RIGHT = (315 * Math.PI) / 180,
+}
+
+export const Weapons = {
```

Comment:
I would move Weapons and Characters into its own class, something like entities.ts, or a separate file for each entity type.

For more longterm planning, I suggest you look into https://en.m.wikipedia.org/wiki/Entity_component_system for game object design ideas.

Reply:
I kept constants.ts but moved characters to entities.ts and weapons to components.ts 5e3c6e863ff1f54563dfbc133dce34867157f64e